src/utils/validation.ts

```
@@ -0,0 +1,137 @@
+    if (value === null) return 'null';
+    if (value === undefined) return 'undefined';
+    if (typeof value === 'number' || typeof value === 'boolean') {
+      return String(value);
+    }
+    if (typeof value === 'object') {
+      return '[object]';
+    }
+    // For functions and other types, return a safe representation
+    return `[${typeof value}]`;
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)